src/components/FormTodo.tsx

```diff
@@ -0,0 +1,85 @@
+import { useEffect, useRef, useState } from 'react';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  onSubmit: (todo: Todo) => Promise<void>;
+  userId: number;
+  onError: (error: string) => void;
+  onSubmitTempTodo: (todo: Todo | null) => void;
+  processingIds: number[];
+};
+
+export const NewTodoForm: React.FC<Props> = ({
+  onSubmit,
+  userId,
+  onError,
+  onSubmitTempTodo,
+  processingIds,
+}) => {
+  const [title, setTitle] = useState('');
+  const [isSubmiting, setIsSubmiting] = useState(false);
+
+  const titleField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    titleField.current?.focus();
+  }, [processingIds.length]);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const normalisedTitle = title.trim();
+
+    if (!normalisedTitle) {
+      onError('Title should not be empty');
+
+      return;
+    }
+
+    setIsSubmiting(true);
+
+    const tempTodo = { userId, completed: false, title: normalisedTitle };
+
+    const newTodo = {
+      userId,
+      completed: false,
+      title: normalisedTitle,
+      id: +new Date(),
```

Comment:
The server adds the ID automatically, so it's redundant.

Reply:
> The server adds the ID automatically, so it's redundant.

Good point

Reply:
I use id to control the focus of the input in the FormTodo component. To avoid creating a new state, I use this state:  
const [processingIds, setProcessingIds] = useState<number[]>([]);